react/Media.tsx

```diff
@@ -0,0 +1,146 @@
+import React from 'react'
+import { Image } from 'vtex.store-image'
+import { Video } from 'vtex.store-video'
+import { useCssHandles, useCustomClasses } from 'vtex.css-handles'
+import type { CssHandlesTypes } from 'vtex.css-handles'
+
+import type {
+  BlockModeProps,
+  ImageModeProps,
+  MediaProps,
+  VideoModeProps,
+} from './MediaTypes'
+
+const IMAGE_CSS_HANDLES = ['imageElement', 'imageElementLink'] as const
+const VIDEO_CSS_HANDLES = [
+  'videoContainer',
+  'videoElement',
+  'fallbackContainer',
+  'fallbackImage',
+  'controlsContainer',
+  'playButton',
+  'trackContainer',
+  'trackTimer',
+  'trackBar',
+  'fullscreenButton',
+  'volumeContainer',
+  'volumeSlider',
+  'volumeButton',
+] as const
+
+// Regex101 won't work if the the escape char is removed
+// https://stackoverflow.com/a/36991065/6352590
+// eslint-disable-next-line no-useless-escape
+const isVideoFileRegex = /((?:https?(?:%3A%2F%2F|:\/\/))(?:www\.)?(?:\S+)(?:%2F|\/)(?:(?!\.(?:mp4|mkv|wmv|m4v|mov|avi|flv|webm))[^\/])*\.(mp4|mkv|wmv|m4v|mov|avi|flv|webm))(?!\/|\.[a-z]{1,3})/
+
+// Regex101 won't work if the the escape char is removed
+// https://stackoverflow.com/a/50777192/6352590
+// eslint-disable-next-line no-useless-escape
+const isVimeoRegex = /(http|https)?:\/\/(www\.|player\.)?vimeo\.com\/(?:channels\/(?:\w+\/)?|groups\/([^\/]*)\/videos\/|video\/|)(\d+)(?:|\/\?)/
+
+// Regex101 won't work if the the escape char is removed
+// https://stackoverflow.com/a/61033353/6352590
+// eslint-disable-next-line no-useless-escape
+const isYoutubeRegex = /(?:https?:\/\/)?(?:www\.)?youtu(?:\.be\/|be.com\/\S*(?:watch|embed)(?:(?:(?=\/[^&\s\?]+(?!\S))\/)|(?:\S*v=|v\/)))([^&\s\?]+)/
+
+function isVideoSrc(src?: string) {
+  if (
+    !src ||
+    !(
+      isVideoFileRegex.test(src) ||
+      isYoutubeRegex.test(src) ||
+      isVimeoRegex.test(src)
+    )
```

Comment:
What do you think about creat a predicate, something like that:

```tsx
function isValidSource(src: string) {
   return isVideoFileRegex.test(src) ||
      isYoutubeRegex.test(src) ||
      isVimeoRegex.test(src)
}
```
 ```tsx
    !isValidSource(src)
```

Reply:
My bad, my suggestion does the same thing your function already does. Ignore it.

Reply:
I think Igor's suggestion makes it clearer. A variable abode the if clause would be kind of the same because I need to assure `src` is not `before` testing it against the regexes.

Reply:
I did a mix of both 😝